### DISCUSSION TITLE
How to allow the player to rename their character?

### DISCUSSION THREAD

**Discussion by godofgrunts** [dialogic-godot/dialogic#875] [Q&A] [answered]
Is there a built in way to do this or do I need to create a new scene with a LineEdit node?

Is it possible to make any character's name a variable instead?

**Comment by zaknafean** [comment, marked as answer]
You would need to build your own input UI to capture user input like that.

And you can make a Dialogic definition to store that result in and use the definition anywhere in Dialogic. 

In addition there seems to be a bug if you're using the DialogNode directly rather than instantiating it via code (**Dialogic.start(timeline_name)**). So be sure to start the Dialog via code!~!

**Reply by coppolaemilio** [reply]
Just to clarify, the name of a character could be the value of a Dialogic definition:
<img width="328" alt="image" src="https://user-images.githubusercontent.com/2206700/166058993-a73acf11-384f-44c0-aabc-b36a57540eac.png">

Character:
<img width="319" alt="image" src="https://user-images.githubusercontent.com/2206700/166059136-2b1286f8-e070-42aa-a380-1fd2e3523b95.png">

In game:
<img width="356" alt="image" src="https://user-images.githubusercontent.com/2206700/166059242-ae461e07-d54b-4ab7-83ca-624e8954373d.png">

**Comment by godofgrunts** [comment]
Okay so I have a problem.

Once I set the variable it doesn't change.

For example. I have [MCName] set to John at the beginning. I then ask the player to rename the character.

I change_scene to new node that has a LineEdit function. When the text is entered I'm calling:

```
func _on_Input_text_entered(new_text):
	print("Name before: %s" % Dialogic.get_variable("MCName"))
	Dialogic.set_variable("MCName", new_text)
	print("Name after: %s" % Dialogic.get_variable("MCName"))
	get_tree().change_scene("res://src/scenes/chapter1/Chapter1.tscn")
```

But when [MCName] speaks, it is still set to John.

**Comment by godofgrunts** [comment]
Removing the answered tag since the solution doesn't actually work.

**Comment by godofgrunts** [comment]
Okay I figured out the issue. My code was working well, but you can't call Dialogic from the Dialogic node if you change the variables.

Calling it from code works fine.
